### ISSUE TITLE
H743 and H743 DFSDM offsets wrong

### ISSUE THREAD

**Issue by David-OConnor** [stm32-rs/stm32-rs#636]
The offsets appear to be correct for all other H743 variants. They all appear to be wrong on H743 and H743V. Working on a fix, but can't get changes to take ATM.

Attempting to apply a common patch of the form:

```yaml
DFSDM:
  _modify:
    DFSDM_CHCFG0R1: 
      addressOffset: 0x00
    DFSDM_CHCFG0R2: 
      addressOffset: 0x04
    DFSDM_AWSCDR0R:
      addressOffset: 0x08
    DFSDM0_DAT0R:
      addressOffset: 0x0C
    DFSDM0_DATIN0R:
      addressOffset: 0x10

    DFSDM_CHCFG1R1: 
      addressOffset: 0x20
    DFSDM_CHCFG1R2: 
      addressOffset: 0x24
    DFSDM_AWSCDR1R:
      addressOffset: 0x28
    DFSDM0_DAT1R:
      addressOffset: 0x2C
    DFSDM0_DATIN1R:
      addressOffset: 0x30.
..
```

Additionally, the SVD names don't match the form of the RM. Eg `FLT0CR2` vs ` DFSDM0_CR2` and `CH2CFGR1` vs `DFSDM_CHCFG0R1`

Also, indexed channels and filters would be nice, eg H7's DMA streams. I'm going to take a stab at the wrong addresses. Any ideas why the fixes I put in aren't taking?

**Comment by adamgreig** [comment]
Could you push a branch that shows them not working? It's hard to tell from just that snippet, which looks OK in isolation.

**Comment by David-OConnor** [comment]
Done - https://github.com/stm32-rs/stm32-rs/pull/637

**Comment by David-OConnor** [comment]
I think the "not working" is due to typos in the reg names I added.

**Comment by adamgreig** [comment]
If nothing else, we usually strip the peripheral name from field names using the `_strip` command for things like this ([example](https://github.com/stm32-rs/stm32-rs/blob/master/devices/common_patches/sdio_f446.yaml#L12)).

**Comment by David-OConnor** [comment]
Added strip, and the address offsets appear to have taken. Leaving as draft until further QC and ops testing. Appreciate it!

**Comment by adamgreig** [comment]
Nice! Annoyingly it seems like the names such as `DFSDM3_CR2` get turned into plain `3_CR2` which is not valid (cannot start with a number). Looks like that's only an issue on 743/743v, so perhaps that patch file could also rename those registers to the `FLT3CR2` name used elsewhere.

**Comment by David-OConnor** [comment]
That would be idea; make it all match RM.

Of note, it seems like it's prefixing `_` to make it valid, but I'm running into something very weird where it's expecting 2 of the regs (`_1_FCR` and `_3_FCR` to be methods (?) based on compile errors, and being ommitted from the top-of-the-module list of offsets. Not sure what's up. Could just make all the names match the RM.

**Comment by adamgreig** [comment]
`svd2rust` will prefix `_`, but the generated SVD file fails validation against the SVD specification because of the invalid names. Since you're already having to put the new address offset for the affected registers, you could also put `name:` in that same block to rename them.

If it's turning registers into methods, that's typically because they overlap or have the same address (like CCMR input/output mode registers on STM32 timers), so they are taken out of the register block and the methods are used instead.

**Comment by David-OConnor** [comment]
Names fixed across H7

**Comment by adamgreig** [comment]
Fixed by #637
